sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Configs.java

```diff
@@ -387,12 +387,12 @@ public static String getNonIdempotentWriteRetryPolicy() {
     public static int getDefaultHttpPoolSize() {
         String valueFromSystemProperty = System.getProperty(HTTP_DEFAULT_CONNECTION_POOL_SIZE);
         if (valueFromSystemProperty != null && !valueFromSystemProperty.isEmpty()) {
-            return Integer.valueOf(valueFromSystemProperty);
+            return Integer.parseInt(valueFromSystemProperty);
```

Comment:
The boxed parsing methods were being used when returning a primitive. The boxed methods should be used sparingly as they don't guard against nullness inputs (except for Boolean) and most the implementations of the boxed parse methods are wrappers around the primitive parsing methods.

Reply:
Agreed, thanks!